pkg/server/plugin/bundlepublisher/exec/exec.go

```diff
@@ -0,0 +1,203 @@
+package exec
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire-plugin-sdk/pluginsdk/support/bundleformat"
+	bundlepublisherv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/bundlepublisher/v1"
+	"github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/types"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/pluginconf"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	pluginName = "exec"
+)
+
+func BuiltIn() catalog.BuiltIn {
+	return builtin(New())
+}
+
+func New() *Plugin {
+	return newPlugin()
+}
+
+// Config holds the configuration of the plugin.
+type Config struct {
+	Cmd    []string `hcl:"cmd" json:"cmd"`
+	Format string   `hcl:"format" json:"format"`
+
+	// bundleFormat is used to store the content of Format, parsed
+	// as bundleformat.Format.
+	bundleFormat bundleformat.Format
+}
+
+func buildConfig(coreConfig catalog.CoreConfig, hclText string, status *pluginconf.Status) *Config {
+	newConfig := new(Config)
+
+	if err := hcl.Decode(newConfig, hclText); err != nil {
+		status.ReportErrorf("unable to decode configuration: %v", err)
+		return nil
+	}
+
+	if len(newConfig.Cmd) < 1 {
+		status.ReportError("configuration is missing cmd")
```

Comment:
If there is no cmd should we exit and return error?

Reply:
Should be fixed